### DISCUSSION TITLE
Support for ordinal classification or regression-based models in the pipeline

### DISCUSSION THREAD

**Discussion by zachschillaci27** [explosion/spaCy#10369] [New Features & Project Ideas]
The TextCategorizer component is ill suited in cases where there is an ordering to the classes or a continuous variable to be predicted. Therefore, support could be added for a TextOrdinalCategorizer or TextRegressor pipeline to predict 
an ordinal classification or a continuous response variable, respectively. Such a pipeline could be used for predicting classic ordinal classification problems such as movie star ratings or regressor problems such as sentiment analysis.

**Comment by koaning** [comment]
Part of me wonders if you'd really need a separate categorizer for this. In the case of a 5-star review rating, you can normalize the values of 1-5 to a single classifier that needs to predict a value between [0, 1]. It's a trick that I use [here](https://github.com/koaning/sentimany/blob/main/sentimany/sentiment/_huggingface.py). I've not tried this approach using spaCy, but conceptually it might be simpler to first re-use the existing tools.
